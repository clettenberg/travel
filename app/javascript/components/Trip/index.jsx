import React from 'react'
import PropTypes from 'prop-types'
import Paper from '@material-ui/core/Paper'

import Map from '../Map'
import PaperList from '../shared/PaperList'
import ThingToList from '../../models/ThingToList'

import styles from './style.module.scss'

const propTypes = {
  trip: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    center: PropTypes.arrayOf(PropTypes.number),
    places: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      point: PropTypes.arrayOf(PropTypes.number)
    }))
  }).isRequired
}

const Trip = ({ trip }) => {
  const points = trip.places.map(place => place.point).filter(Boolean)
  const thingsToList = trip.places.map(({ id, name }) => (
    new ThingToList(`/places/${id}`, name, id)
  ))

  return (
    <React.Fragment>
      <div className={styles.container}>
        <Map
          points={points}
          center={trip.center}
          zoom={3}
        />
      </div>
      <Paper className={styles.content}>
        <PaperList
          title={trip.title}
          actionButtonUrl={`/trips/${trip.id}/places/new`}
          actionButtonText='Add Place'
          thingsToList={thingsToList}
        />
      </Paper>
    </React.Fragment>
  )
}

Trip.propTypes = propTypes
export default Trip
