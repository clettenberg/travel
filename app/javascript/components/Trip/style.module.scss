@import '../shared/breakpoints.scss';

$map-content-split-percentage: 60%;

.container {
  div[class*=this-is-the-map] {
    position: absolute;
    top: 0;
    bottom: 0;
    width: 100%;
  }

  @media (max-width: $mobile-breakpoint) {
    position: absolute;
    top: 0;
    bottom: $map-content-split-percentage;
    width: 100%;
  }
}

.content {
  position: absolute;
  background: white;
  margin: 10px;
  max-height: 90vh;
  width: 400px;
  overflow: scroll;

  @media (max-width: $mobile-breakpoint) {
    bottom: 0;
    padding: 0;
    margin: 0;
    height: $map-content-split-percentage;
    min-width: 100%;
  }
}
