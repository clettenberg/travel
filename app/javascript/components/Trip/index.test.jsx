
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Trip from './index'

describe('Trip', () => {
  it('should render', () => {
    const wrapper = shallow(
      <Trip trip={{
        id: 26,
        title: 'UK with Ron and Stacey',
        places: [{
          id: 286,
          name: 'The Pink House (Behind the caravan park)',
          point: [-5.21714480341615, 51.8780559],
          note: '',
          startDate: '09/20/19',
          endDate: '09/22/19'
        }, {
          id: 287,
          name: 'AirBnB in Edinburgh',
          point: [-3.1936512, 55.9499331],
          note: 'u003cdivu003eGreat flat in old Edinburghu003c/divu003e',
          startDate: '09/25/19',
          endDate: '09/28/19'
        }],
        center: [-2.6707358336225657, 53.941955750000005],
        startDate: '09/15/19',
        endDate: '09/28/19'
      }}
      />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
