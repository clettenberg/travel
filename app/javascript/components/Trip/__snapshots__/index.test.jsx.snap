// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip should render 1`] = `
<Fragment>
  <div>
    <Map
      center={
        Array [
          -2.6707358336225657,
          53.941955750000005,
        ]
      }
      mapClassName="this-is-the-map"
      points={
        Array [
          Array [
            -5.21714480341615,
            51.8780559,
          ],
          Array [
            -3.1936512,
            55.9499331,
          ],
        ]
      }
      zoom={3}
    />
  </div>
  <WithStyles(ForwardRef(Paper))>
    <PaperList
      actionButtonText="Add Place"
      actionButtonUrl="/classic/trips/26/places/new"
      thingsToList={
        Array [
          ThingToList {
            "date": "09/20/19 - 09/22/19",
            "id": 286,
            "title": "The Pink House (Behind the caravan park)",
            "url": "/places/286",
          },
          ThingToList {
            "date": "09/25/19 - 09/28/19",
            "id": 287,
            "title": "AirBnB in Edinburgh",
            "url": "/places/287",
          },
        ]
      }
      title="UK with Ron and Stacey"
    />
  </WithStyles(ForwardRef(Paper))>
</Fragment>
`;
